test/jdk/jdk/lambda/separate/ClassToInterfaceConverter.java

```diff
@@ -23,75 +23,64 @@
+import static jdk.internal.classfile.Opcode.*;
 
 public class ClassToInterfaceConverter implements ClassFilePreprocessor {
 
-    private String whichClass;
+    private final String whichClass;
 
     public ClassToInterfaceConverter(String className) {
         this.whichClass = className;
     }
 
-    private boolean utf8Matches(ClassFile.CpEntry entry, String v) {
-        if (!(entry instanceof ClassFile.CpUtf8)) {
-            return false;
-        }
-        ClassFile.CpUtf8 utf8 = (ClassFile.CpUtf8)entry;
-        if (v.length() != utf8.bytes.length) {
-            return false;
-        }
-        for (int i = 0; i < v.length(); ++i) {
-            if (v.charAt(i) != utf8.bytes[i]) {
-                return false;
-            }
-        }
-        return true;
-    }
-
-    private void convertToInterface(ClassFile cf) {
-        cf.access_flags = 0x0601; // ACC_INTERFACE | ACC_ABSTRACT | ACC_PUBLIC
-        ArrayList<ClassFile.Method> new_methods = new ArrayList<>();
-        // Find <init> method and delete it
-        for (int i = 0; i < cf.methods.size(); ++i) {
-            ClassFile.Method method = cf.methods.get(i);
-            ClassFile.CpEntry name = cf.constant_pool.get(method.name_index);
-            if (!utf8Matches(name, "<init>")) {
-                new_methods.add(method);
-            }
-        }
-        cf.methods = new_methods;
-        //  Convert method tag. Find Methodref, which is not "<init>" and only invoked by other methods
-        //  in the interface, convert it to InterfaceMethodref
-        ArrayList<ClassFile.CpEntry> cpool = new ArrayList<>();
-        for (int i = 0; i < cf.constant_pool.size(); i++) {
-            ClassFile.CpEntry ce = cf.constant_pool.get(i);
-            if (ce instanceof ClassFile.CpMethodRef) {
-                ClassFile.CpMethodRef me = (ClassFile.CpMethodRef)ce;
-                ClassFile.CpNameAndType nameType = (ClassFile.CpNameAndType)cf.constant_pool.get(me.name_and_type_index);
-                ClassFile.CpEntry name = cf.constant_pool.get(nameType.name_index);
-                if (!utf8Matches(name, "<init>") && cf.this_class == me.class_index) {
-                    ClassFile.CpInterfaceMethodRef newEntry = new ClassFile.CpInterfaceMethodRef();
-                    newEntry.class_index = me.class_index;
-                    newEntry.name_and_type_index = me.name_and_type_index;
-                    ce = newEntry;
-                }
-            }
-            cpool.add(ce);
-        }
-        cf.constant_pool = cpool;
+    private byte[] convertToInterface(ClassModel classModel) {
+        return Classfile.of().build(classModel.thisClass().asSymbol(),
+                classBuilder ->  {
+                    for (ClassElement ce : classModel) {
+                        if (ce instanceof AccessFlags accessFlags) {
+                            classBuilder.withFlags(0x0601); // ACC_INTERFACE | ACC_ABSTRACT | ACC_PUBLIC);
```

Comment:
We can just use `withFlags(Classfile.ACC_INTERFACE | Classfile.ACC_ABSTRACT | Classfile.ACC_PUBLIC)` as it's a constant expression and inlined by javac. And the `accessFlags` variable above appears unused.

Reply:
I have same thought too.  I have that changed in my local repo but not have the chance to push it.